app/src/main/cpp/skyline/services/nvdrv/devices/nvmap.cpp

```diff
@@ -1,172 +1,147 @@
-// SPDX-License-Identifier: MPL-2.0
-// Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
+// SPDX-License-Identifier: MIT OR MPL-2.0
+// Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
 
+#include <services/nvdrv/devices/deserialisation/deserialisation.h>
+#include <services/nvdrv/core/nvmap.h>
 #include "nvmap.h"
 
 namespace skyline::service::nvdrv::device {
-    NvMap::NvMapObject::NvMapObject(u32 id, u32 size) : id(id), size(size) {}
+    NvMap::NvMap(const DeviceState &state, Core &core, const SessionContext &ctx) : NvDevice(state, core, ctx) {}
+
+    PosixResult NvMap::Create(In<u32> size, Out<NvMapCore::Handle::Id> handle) {
+        auto h{core.nvMap.CreateHandle(util::AlignUp(size, PAGE_SIZE))};
+        if (h) {
+            (*h)->origSize = size; // Orig size is the unaligned size
+            handle = (*h)->id;
+            state.logger->Debug("handle: {}, size: 0x{:X}", (*h)->id, size);
+        }
+
+        return h;
+    }
 
-    NvMap::NvMap(const DeviceState &state) : NvDevice(state) {}
+    PosixResult NvMap::FromId(In<NvMapCore::Handle::Id> id, Out<NvMapCore::Handle::Id> handle) {
+        state.logger->Debug("id: {}", id);
 
-    NvStatus NvMap::Create(IoctlType type, span<u8> buffer, span<u8> inlineBuffer) {
-        struct Data {
-            u32 size;   // In
-            u32 handle; // Out
-        } &data = buffer.as<Data>();
+        // Handles and IDs are always the same value in nvmap however IDs can be used globally given the right permissions.
+        // Since we don't plan on ever supporting multiprocess we can skip implementing handle refs and so this function
+        // just does simple validation and passes through the handle id.
```

Comment:
Same issue with comment spacing here, only break for text where there would be a new paragraph or *at the very least* a new sentence. Breaking mid-sentence looks horrible.